### PR TITLE
Fix GraphMatch random_state behavior with n_init > 1, parallelize

### LINKED ISSUE #761: Parallelize over initializations in `GraphMatch`
## Is your feature request related to a problem? Please describe.
Graph matching can take a while 

## Describe the solution you'd like
Would like parallelization over each initialization in graph matching using joblib. Should be a one-liner, basically

### LINKED ISSUE #765: [BUG] Fix random state behavior with multiple inits in GraphMatch
## Expected Behavior
Even when passing a `random_state` to `GraphMatch`, each init (child process) should get it's own random state 

## Actual Behavior
Each one gets the same

## Additional Details
Joblib has a nice tutorial on "differently random, but deterministic" https://joblib.readthedocs.io/en/latest/auto_examples/parallel_random_state.html#fixing-the-random-state-to-obtain-deterministic-results

We might as well take care of this and #761 at the same time?
cc @asaadeldin11 lmk if you agree that we might as well do both at once?

### PR DESCRIPTION
**Pull request by asaadeldin11**
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [x] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate?

#### Reference Issues/PRs
Closes #761
Closes #765.

#### What does this implement/fix? Briefly explain your changes.
Fixes bug where random_state when n_init > 1 was always the same. Also parallelizes over inits

### CODE REVIEW

File: graspologic/match/gmp.py [hunk 1]
```diff
@@ -228,17 +232,33 @@ def fit(self, A, B, seeds_A=[], seeds_B=[], S=None):
...
             "shuffle_input": self.shuffle_input,
             "maxiter": self.max_iter,
             "tol": self.eps,
         }
-
-        res = min(
-            [quadratic_assignment(A, B, options=options) for i in range(self.n_init)],
-            key=lambda x: x.fun,
-        )
+        if self.n_init > 1:
```
Comment (bdpedigo) [inline graspologic/match/gmp.py:240]: @asaadeldin11 I don't even think we need this if/else, thoughts?

Comment (bdpedigo) [reply graspologic/match/gmp.py:240]: would make the code simpler is all

**Comment by bdpedigo** [comment]
added a test, feel free to change it

File: graspologic/match/gmp.py (hunk 1 above)
Comment (asaadeldin11) [reply graspologic/match/gmp.py:240]: i had it like that before but the tests were taking way slower

Comment (asaadeldin11) [reply graspologic/match/gmp.py:246]: i think creating a `Parallel` joblib object takes time

Comment (bdpedigo) [reply graspologic/match/gmp.py:246]: interesting, I guess there is a bit of overhead to it. can leave as is for now, maybe add a comment?

Comment (asaadeldin11) [reply graspologic/match/gmp.py:246]: so not worth it if `n_init` = 1 i think

Comment (bdpedigo) [reply graspologic/match/gmp.py:246]: you had `n_jobs` set to -1 tho? what about when `n_jobs` is default 1?

Comment (bdpedigo) [reply graspologic/match/gmp.py:246]: regardless I guess I don't care too much but I thought they were smart enough about that stuff in joblib. maybe not

Comment (asaadeldin11) [reply graspologic/match/gmp.py:246]: i can try again with `n_jobs` set to 1

**Review by bdpedigo — APPROVED** [review]
Think this is good to go, @asaadeldin11 you can decide whether to just leave a comment about efficiency or see if the n_jobs = 1 fix helps performance, otherwise I'm ready to merge

**Comment by bdpedigo** [comment]
@asaadeldin11 somehow that broke old tests, any idea why? maybe the tests just werent that stable to initialization? Or do we think something else is going on?

**Comment by asaadeldin11** [comment]
> @asaadeldin11 somehow that broke old tests, any idea why? maybe the tests just werent that stable to initialization? Or do we think something else is going on?

Strange, they were passing for me locally for me. The padded sgm test makes sense, i remove two nodes and get 48/50 match ratio, so I'll change the 1.0 to 0.95 since there's no reason to expect the isolates to match correctly. 
The second one I'm less sure about, but I think it might be resolved by changed n in `test_sim` from 90 to something larger (120 or 150). Thoughts?

**Comment by bdpedigo** [comment]
> > @asaadeldin11 somehow that broke old tests, any idea why? maybe the tests just werent that stable to initialization? Or do we think something else is going on?
> 
> Strange, they were passing for me locally. The padded sgm test makes sense, i remove two nodes and get 48/50 match ratio, so I'll change the 1.0 to 0.95 since there's no reason to expect the isolates to match correctly.
> The second one I'm less sure about, but I think it might be resolved by changed n in `test_sim` from 90 to something larger (120 or 150). Thoughts?

is there any other way to just make the problem easier?

**Comment by asaadeldin11** [comment]
> is there any other way to just make the problem easier?
I can just make the similarity matrix the identity matrix, and make it an ER graph?

**Pull request merged**